src/pages/ResourcePage.tsx

```diff
@@ -36,79 +40,121 @@ const ResourcePage: FC<RouteProps> = (props) => {
   const [preview, setPreview] = useState<Preview>({ type: '', theSource: '' });
+  const [isLoadingPreview, setIsLoadingPreview] = useState<boolean>(false);
   const [tags, setTags] = useState<string[]>([]);
+  const [isLoadingTags, setIsLoadingTags] = useState<boolean>(false);
   const [licenses, setLicenses] = useState<License[]>([]);
+  const [isLoadingLicenses, setIsLoadingLicenses] = useState<boolean>(false);
 
   useEffect(() => {
-    const collectResourceData = async (identifier: string) => {
-      getResource(identifier).then((response) => {
-        setResource(response.data);
-      });
-      getResourceCreators(identifier).then((response) => {
-        setCreator(response.data);
-      });
-      getResourceContents(identifier).then((response) => {
-        const type = response?.data[0]?.features?.dlr_content_content_type
-          ? response?.data[0]?.features?.dlr_content_content_type
-          : '';
-        setPreview({
-          type,
-          theSource: `${API_URL}${API_PATHS.guiBackendResourcesContentPath}/${response?.data[0]?.identifier}/delivery?jwt=${localStorage.token}`,
+    const fetchData = async (identifier: string) => {
+      getResource(identifier)
+        .then((response) => {
+          setResource(response.data);
+        })
+        .finally(() => {
+          setIsLoadingResource(false);
+        });
+      getResourceCreators(identifier)
+        .then((response) => {
+          setCreators(response.data);
+        })
+        .finally(() => {
+          setIsLoadingCreator(false);
+        });
+      getResourceContents(identifier)
+        .then((response) => {
+          const type = response?.data[0]?.features?.dlr_content_content_type
+            ? response?.data[0]?.features?.dlr_content_content_type
+            : '';
+          setPreview({
+            type,
+            theSource: `${API_URL}${API_PATHS.guiBackendResourcesContentPath}/${response?.data[0]?.identifier}/delivery?jwt=${localStorage.token}`,
+          });
+        })
+        .finally(() => {
+          setIsLoadingPreview(false);
+        });
+      getResourceTags(identifier)
+        .then((response) => {
+          setTags(response.data);
+        })
+        .finally(() => {
+          setIsLoadingTags(false);
+        });
+      getResourceLicenses(identifier)
+        .then((response) => {
+          setLicenses(response.data);
+        })
+        .finally(() => {
+          setIsLoadingLicenses(false);
         });
-      });
-      getResourceTags(identifier).then((response) => {
-        setTags(response.data);
-      });
-      getResourceLicenses(identifier).then((response) => {
-        setLicenses(response.data);
-      });
     };
-
     if (identifier) {
       setIsLoadingResource(true);
-      collectResourceData(identifier).finally(() => {
-        setIsLoadingResource(false);
-        //todo: presenter siden gradvis ?
-      });
+      setIsLoadingLicenses(true);
```

Comment:
trenger vi waitere for hvert endepunkt ? Vi kunne hentet hele posten med sine kall slik vi gjør med user - og så pakket den inn i en "promise all"  - ulempen er jo hvis en av tjenesten krasjer, da ....

Reply:
Skal jeg endre det tilbake slik det var før? da ventet det på at alle resursene var ferdig får setIsLoading ble satt til false?

Reply:
jeg foreslår vi kan beholde waiterene, men bytte ut 
`      {(isLoadingResource || isLoadingTags || isLoadingCreator || isLoadingPreview) && <CircularProgress />}
`
med
      `{(isLoadingResource && <CircularProgress />}`
slik at vi bare har spinneren på lasting av selve ressursen - så kan vi evt legge på spinnere på hver seksjon som lastes for seg senere.
Vi burde vel også sett på placeholdere slik at siden ikke "hopper" for mye når den laster

Reply:
Sånn fikset. 
Har lagd det med Promise all. Forøvrig så vil man få opp elementer etterhvert som de er lastet inn, men circular progress vil fortsette å snurre til alle tingene er lastet ned. Det vil vel sikre hvis deler av backenden ikke fungerer?